config/externalname.go

```diff
@@ -67,6 +67,10 @@ var TerraformPluginFrameworkExternalNameConfigs = map[string]config.ExternalName
 	"aws_opensearchserverless_security_policy": config.NameAsIdentifier,
 	// VPCEndpoint can be imported using the AWS-assigned VPC Endpoint ID, i.e. vpce-0a957ae9ed5aee308
 	"aws_opensearchserverless_vpc_endpoint": opensearchserverlessVpcEndpoint(),
+
+	// CodeGuru Profiler
+	// Profiling Group can be imported using the profiling group name
+	"aws_codeguruprofiler_profiling_group": config.NameAsIdentifier,
```

Comment:
Apparently, terraform doesn't allow the id to be empty. Since the id of the resource is `id - Name of the profiling group.`, we encounter the error `profilingGroupName must not be empty`. 

There are similar external name functions we use in such cases, for instance [cognitoUserPoolClient()](https://github.com/crossplane-contrib/provider-upjet-aws/blob/285b5e22ee25fdb6d95c8d445e539298dbcf2ce6/config/externalname.go#L2735C6-L2735C29)

- The following configuration will work:

```suggestion
	"aws_codeguruprofiler_profiling_group": config.codeguruprofilerProfilingGroup(),
```

- You can add the following function to the end of the `externalname.go` file:


```
func codeguruprofilerProfilingGroup() config.ExternalName {
	e := config.IdentifierFromProvider
	e.GetIDFn = func(ctx context.Context, externalName string, parameters map[string]interface{}, cfg map[string]interface{}) (string, error) {
		if externalName == "" {
			return "invalidnonemptystring", nil
		}
		return externalName, nil
	}
	return e
}
```

Reply:
Thanks, fixed! I see the terraform provider sets state.ID = profilingGroup.Name. I don't fully understand the pros and cons here (i.e. what is broken with using NameAsIdentifier). One thing is that we now both have name and external-name, but I think they will have to be the same to make any sense. Maybe we should have a cel-expression in the future to enforce that if external name is set, it must match the name?